frontend/src/lib/components/FlowPreviewContent.svelte

```diff
@@ -98,6 +103,21 @@
 		isRunning = false
 	}
 
+	$: {
+		if (selectedJobStep !== undefined && job?.flow_status?.modules !== undefined) {
+			selectedJobStepIsTopLevel =
+				job?.flow_status?.modules.map((m) => m.id).indexOf(selectedJobStep) >= 0
+			let moduleDefinition = job?.raw_flow?.modules.find((m) => m.id == selectedJobStep)
+			if (moduleDefinition?.value.type == 'forloopflow') {
+				selectedJobStepType = 'forloop'
+			} else if (moduleDefinition?.value.type == 'branchall') {
+				selectedJobStepType = 'branchall'
+			} else {
+				selectedJobStepType = 'single'
+			}
+		}
+	}
```

Comment:
better practice to make it a function whose reactive dependencies are taken as arg, otherwise it's hard to read what this is re-evaluated on

Reply:
I see what you mean, I'll update 👍